File: Typecast/TCCharacterMapListViewController.m
//
//  TCCharacterMapListViewController.m
//  Type Designer
//
//  Created by David Schweinsberg on 22/12/12.
//  Copyright (c) 2012 David Schweinsberg. All rights reserved.
//

#import "TCCharacterMapListViewController.h"
#import "TCCharacterMapWindowController.h"
#import "Type_Designer-Swift.h"

@interface TCCharacterMapListViewController ()

@end

@implementation TCCharacterMapListViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Initialization code here.
    }
    
    return self;
}

- (void)showCharacterMaps:(NSArray *)maps
{
    TCCharacterMapWindowController *windowController = [[TCCharacterMapWindowController alloc] initWithWindowNibName:@"CharacterMapWindow"];
    [[self document] addWindowController:windowController];
    [windowController setCmapIndexEntry:[maps lastObject]];
    [windowController showWindow:self];
}

@end

File: Typecast/TCCharacterMapListViewController.swift
//
//  TCCharacterMapListViewController.swift
//  Type Designer
//
//  Created by David Schweinsberg on 8/7/17.
//  Copyright © 2017 David Schweinsberg. All rights reserved.
//

import Cocoa

class TCCharacterMapListViewController: NSViewController {
  weak var document: FontDocument?

  override func viewDidLoad() {
    super.viewDidLoad()
    // Do view setup here.
  }

  func showCharacterMaps(_ maps: [TCCmapIndexEntry]) {
    let windowController = TCCharacterMapWindowController(windowNibName: NSNib.Name(rawValue: "CharacterMapWindow"))
    document?.addWindowController(windowController)
    windowController.cmapIndexEntry = maps.last
    windowController.showWindow(self)
  }
}
